payments/models.py

```diff
@@ -57,52 +66,102 @@ def external_account(self, bank_account):
 
 class StripeEvent(models.Model):
     event_id = models.CharField(primary_key=True, max_length=100, blank=True)
+    user_id = models.CharField(max_length=100, blank=True)
     type = models.CharField(max_length=100, blank=True)
     message_text = models.TextField()
+    verified = models.BooleanField(default=False)
     processed = models.BooleanField(default=False)
+    created = models.DateTimeField(null=True, blank=True)
 
+    @property
     def message(self):
         return json.loads(self.message_text)
 
     def save(self, *args, **kwargs):
+        self.created = timezone.now()
+        super(StripeEvent, self).save(*args, **kwargs)
+
         try:
-            webhooks[self.type].process(self)
+            self.user_id = self.message_text['user_id']
+            retrieved_event = stripe.Event.retrieve(
+                id=self.event_id, stripe_account=self.user_id)
         except KeyError:
+            retrieved_event = stripe.Event.retrieve(id=self.event_id)
+
+        if retrieved_event:
+            self.verified = True
+            self.type = retrieved_event['type']
+            self.message_text = json.dumps(retrieved_event)
+        super(StripeEvent, self).save(*args, **kwargs)
+
+        # attempt to process a webhook for the event type
+        try:
+            webhooks[self.type](event=self).process()
+            self.processed = True
+        except KeyError:
+            # TODO: Add something like papertrail to log this
             pass
         super(StripeEvent, self).save(*args, **kwargs)
 
 webhooks = {}
 
 
-class AccountUpdatedProcessor(object):
-    def process(self, event):
-        message = event.message()
-        account_id = message['user_id']
+class WebHookProcessor(object):
+
+    def __init__(self, event, *args, **kwargs):
+        self.object = event.message['data']['object']
+        super(WebHookProcessor, self).__init__(*args, **kwargs)
+
+    def process():
+        raise NotImplementedError
+
+
+class AccountUpdatedProcessor(WebHookProcessor):
+    def process(self):
         try:
-            account = StripeAccount.objects.get(account_id=account_id)
+            account = StripeAccount.objects.get(account_id=self.account_id)
```

Comment:
I think this is supposed to be `self.object['account_id']` or `self.object.account_id`?